### PR TITLE
feat(nuxi): manage modules via cli

### PR DESCRIPTION
**Pull request by Intevel**
### 🔗 Linked issue

https://github.com/nuxt/rfcs/issues/31
https://github.com/nuxt/framework/discussions/569
#3915

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

With the new commands you can now add different Nuxt 3 modules (& bridge) to the project via Nuxi. The given module name is matched with a list that is currently still done manually (maybe in the future an API?), the name is matched to prevent that not Nuxt modules are installed. The module is added to the `.nuxtrc` file which is merged with the normal nuxt config.

carbon

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Intevel** [comment]
@Baroshem

**Comment by pi0** [comment]
Thanks for the PR @Intevel. Some notes:

- Please disable local linter (prettier) and run `yarn lint --fix`
- We can start with `nuxi module add <npm name>` syntax (later integrate with [nuxt modules](https://github.com/nuxt/modules) DB but is not straight forward)
- `lmify` is deprecated. Working on a new package. It is not safe to rely on.

**Comment by Intevel** [comment]
Thank you @pi0. 
I had already talked with @Atinux that we draft this PR for now. I fixed the linting & disabled prettier. Currently have some kind of manual DB inserted with names, similar to slugs. That would be just an idea.

File: packages/nuxi/src/commands/module.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      process.exit(1)
+    }
+    if (!name) {
+      consola.error('Name argument is missing!')
+      process.exit(1)
+    }
+
+    // Checking if provided package is a Nuxt Module
+    const module = await findModule(name)
+    if (!module) { consola.error('Please check the module list, and ensure if') }
```
Comment (Baroshem) [inline packages/nuxi/src/commands/module.ts:32]: Is this error message correct? It seems to end without proper description 😉

File: packages/nuxi/src/utils/modules.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    packageName: 'nuxt-hue'
+  },
+  {
+    name: 'grapqhl',
+    packageName: 'nuxt-graphql-client'
+  }
+]
+
+export async function findModule (name: string) {
+  const modules = availableModules.filter(
```
Comment (Baroshem) [inline packages/nuxi/src/utils/modules.ts:77]: Maybe you can use find method instead?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: packages/nuxi/src/utils/modules.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    name: 'nuxt-hue',
+    packageName: 'nuxt-hue'
+  },
+  {
+    name: 'grapqhl',
+    packageName: 'nuxt-graphql-client'
+  }
+]
+
+export async function findModule (name: string) {
```
Comment (Baroshem) [inline packages/nuxi/src/utils/modules.ts:76]: Maybe better here would be to name this method findModulePackageName and just return the module package name? If I am correct only the package name is being used later on

File: packages/nuxi/src/commands/module.ts (hunk 1 above)
Comment (Intevel) [reply packages/nuxi/src/commands/module.ts:32]: I will fix it, thank you 😅

File: packages/nuxi/src/utils/modules.ts (hunk 3 above)
Comment (Intevel) [reply packages/nuxi/src/utils/modules.ts:76]: I think it's too long ^^

**Comment by Atinux** [comment]
Thank you for the PR :)

Agree to start with `nuxi module add <npm name>` since it allow anyone to use new packages that are not officially listed.

**Comment by Intevel** [comment]
Should the `npm name` still be checked against a list to make sure it is a Nuxt module?

**Comment by Intevel** [comment]
> Thanks for the PR @Intevel. Some notes:
> 
> 
> 
> - Please disable local linter (prettier) and run `yarn lint --fix`
> 
> - We can start with `nuxi module add <npm name>` syntax (later integrate with [nuxt modules](https://github.com/nuxt/modules) DB but is not straight forward)
> 
> - `lmify` is deprecated. Working on a new package. It is not safe to rely on. 
> 
> 

@pi0 https://github.com/Intevel/qpind Maybe this instead of `lmify` for this?

**Comment by pi0** [comment]
Nice work on qpind @Intevel 💚 However we already have [antfu/install-pkg](https://github.com/antfu/install-pkg) and other solutions, I would still wait for new unjs package designed for nuxi and nuxt universal integration.

**Comment by Intevel** [comment]
@pi0 Any updates on this?

**Comment by Baroshem** [comment]
Folks, can we start working on this feature again? I think this would be an awesome DX improvement to the Nuxi tool. I can help with implementing this as well :)

**Comment by Intevel** [comment]
Surely, so the package installing can we do with install-pkg but where do we get the data from to check if this is a nuxt module or not?

**Comment by Baroshem** [comment]
Can't we just parse the user input against a regexp and if it includes the `nuxt` keywoard assume that it is a module? @nuxt/@nuxtjs/nuxt-something should all work with that. I think even right now, you will only handle properly the modules that are in the modules array so we can always log to the user (Module not found in the supported modules DB/array, please configure the module by yourself)

**Comment by Intevel** [comment]
@pi0 What do you think? Is a regex in this case the best way?

**Comment by Intevel** [comment]
@Baroshem pushed some new stuff, what are you're thoughts about it? (btw i know its not an regex, wrong commit message)

**Comment by Baroshem** [comment]
@Intevel 

Yes, this is what I meant in my proposition in the Discord channel. Nice work! Now, let's wait for Pio, Daniel, or Sebastien to give their opinion and recommend next steps :)

**Comment by manniL** [comment]
@Intevel Did you close this intentionally?

**Comment by pi0** [comment]
Supporting module management from CLI is definitely one thing we always dreamt of having but has been tricky so far because we lacked the CLI infra and AST modification tool to make it happen.

Would be happy to have you on board in next weeks on this feature @Intevel and sorry was delayed in the roadmap.

**Comment by Baroshem** [comment]
I can help with this as well. Just let me know :)

**Comment by Intevel** [comment]
Nice to hear, thank you.

**Pull request closed without merging**